core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -539,6 +547,8 @@ private void syncRemoteChangeEventToLocal(
       return;
     }
 
+    // TODO(QUESTION FOR REVIEW): This code is not in the spec, but if I don't include it, then
+    // testInsertThenSyncUpdateThenUpdate will fail. Should I add it to the spec?
```

Comment:
It's this in spec:
```
drop the event if the version counter of the remote event less than or equal to the version counter of the local document
```

Reply:
But that's in a different place. The way the spec is written, this check is only made "if the document does not have local writes pending". Does this mean we should reorder the spec so that the check occurs before the "If the document does not have local writes pending" check?

Reply:
I suppose it should be changed in the spec. We do this check if we have changes or not.